packages/cdk-ecs-codedeploy-resources/lib/index.ts

```diff
@@ -0,0 +1,288 @@
+        ],
+        loadBalancerInfo: {
+          targetGroupPairInfoList: [
+            {
+              prodTrafficRoute: {
+                listenerArns: [
+                  props.listenerArn
+                ]
+              },
+              targetGroups: [
+                {
+                  name: `${props.targetGroupName}`
+                },
+                {
+                  name: `${appPrefix}-TargetGroup2`
+                }
+              ]
+            }
+          ]
+        },
+      },
+      physicalResourceId: customresource.PhysicalResourceId.of('deployment-group-custom-resource')
+    }
+
+    let codedeployUpdateDeploymentGroupCall: customresource.AwsSdkCall = {
+      service: 'CodeDeploy',
+      action: 'updateDeploymentGroup',
+      parameters: {
+        applicationName: applicationName,
+        currentDeploymentGroupName: deploymentGroupName,
+        deploymentConfigName: "CodeDeployDefault.ECSAllAtOnce",
+        autoRollbackConfiguration: {
+          enabled: true,
+          events: [
+            "DEPLOYMENT_FAILURE",
+            "DEPLOYMENT_STOP_ON_REQUEST"
+          ]
+        },
+        blueGreenDeploymentConfiguration: {
+          deploymentReadyOption: {
+            actionOnTimeout: "CONTINUE_DEPLOYMENT"
+          },
+          terminateBlueInstancesOnDeploymentSuccess: {
+            action: "TERMINATE",
+            terminationWaitTimeInMinutes: terminationTimeout
+          }
+        },
+        deploymentStyle: {
+          deploymentOption: "WITH_TRAFFIC_CONTROL",
+          deploymentType: "BLUE_GREEN"
+        },
+        ecsServices: [
+          {
+            serviceName: props.serviceName,
+            clusterName: props.clusterName
+          }
+        ],
+        loadBalancerInfo: {
+          targetGroupPairInfoList: [
+            {
+              prodTrafficRoute: {
+                listenerArns: [
+                  props.listenerArn
+                ]
+              },
+              targetGroups: [
+                {
+                  name: `${props.targetGroupName}`
+                },
+                {
+                  name: `${appPrefix}-TargetGroup2`
+                }
+              ]
+            }
+          ]
+        },
+      },
+      physicalResourceId: customresource.PhysicalResourceId.of('deployment-group-update-custom-resource')
+    }
+
+    const codeDeployDeploymentGroup = new customresource.AwsCustomResource(this, 'CodeDeployDeploymentGroup', {
+      onCreate: codedeployCreateDeploymentGroupCall,
+      onUpdate: codedeployUpdateDeploymentGroupCall,
+      onDelete: {
+        service: 'CodeDeploy',
+        action: 'deleteDeploymentGroup',
+        parameters: {
+          applicationName: applicationName,
+          deploymentGroupName: deploymentGroupName
+        }
+      },
+      policy: customresource.AwsCustomResourcePolicy.fromSdkCalls({ resources: customresource.AwsCustomResourcePolicy.ANY_RESOURCE }),
+      role: customResourceIamRole
```

Comment:
You actually shouldn't need to role here. From the [documentation](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_custom-resources.AwsCustomResource.html#role) it looks like it generates a new role anyway based on the required `policy` attribute you already have (which is granting full access) to everything. If you can it would be preferable to only grant the custom resources the permission it needs to make the SDK call you have listed.